### DISCUSSION TITLE
How to implement  close-to-tray funtion?

### DISCUSSION THREAD

**Discussion by Jimex** [lepoco/wpfui#150] [Q&A]
Does anyone know how to implement the function that minimizes the window to the tray after clicking the close icon?

Thank you.

**Comment by laolarou726** [comment]
You can create a NotifyIcon by
```c#

// Configure and show a notification icon in the system tray
_notifyIcon = new NotifyIcon
{
    BalloonTipText = "Text",
    Text = "Text",
    Icon = new Icon("Path/To/Your/Icon.ico"),
    Visible = true
};

_notifyIcon.ShowBalloonTip(1000);

```

**Reply by pomianowski** [reply]
This requires `UseWindowsForms` to be present, it is not a WPF solution.

**Reply by laolarou726** [reply]
I think here is the solution for WPF: [https://github.com/HavenDV/H.NotifyIcon](https://github.com/HavenDV/H.NotifyIcon).

**Reply by pomianowski** [reply]
@laolarou726, there is internal WPF UI solution in progress
https://github.com/lepoco/wpfui/blob/ff70f6aba64e2858b6841eb9ce74c3c4ac413f5b/src/WPFUI/Tray/INotifyIcon.cs#L16

**Reply by Jimex** [reply]
> @laolarou726, there is internal WPF UI solution in progress
> 
> https://github.com/lepoco/wpfui/blob/ff70f6aba64e2858b6841eb9ce74c3c4ac413f5b/src/WPFUI/Tray/INotifyIcon.cs#L16

I guess the best way would be to add a CloseActionOverride to WPFUI like MinimizeActionOverride, so that we can call the MinimizeWindow method and bypass the close event.

public Action<TitleBar, Window> MinimizeActionOverride { get; set; } = null;

**Comment by laolarou726** [comment]
But i think this is the only way to implement a tray icon on wpf.

在 2022年6月10日，01:34，Pomian ***@***.***> 写道：

﻿

This requires UseWindowsForms to be present, it is not a WPF solution.
